### PR TITLE
worker: retry popping before going to sleep

### PR DESCRIPTION
**Pull request by BusyJay**
To avoid frequently sleep and wake up, try again popping several times
before putting thread to sleep.

### CODE REVIEW

File: src/pool/worker.rs [hunk 1]
```diff
@@ -19,21 +20,34 @@ where
...
+        if t.is_some() {
+            self.runner.resume(&mut self.local);
+        }
+        t
+    }
+
     pub fn run(mut self) {
         self.runner.start(&mut self.local);
         while !self.local.core().is_shutdown() {
             let task = match self.local.pop() {
```
Comment (sticnarf) [inline src/pool/worker.rs]: Can we just use the implementation of `slow_pop` here and skip this `pop`?

Comment (BusyJay) [reply src/pool/worker.rs]: Just to avoid additional function jump.

Comment (sticnarf) [reply src/pool/worker.rs]: Even now the compiler seems to inline `slow_pop`. (`slow_pop` does not apppear in symbol list)

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by AndreMouche — APPROVED** [review]
LGTM

**Pull request merged**